lib/api/transformers/allow_resource_versions.ex

```diff
@@ -0,0 +1,21 @@
+defmodule AshPaperTrail.Api.Transformers.AllowResourceVersions do
+  @moduledoc """
+  Adds any version resources to the api for any resources. An alternative to
+  AshPaperTrail.Api.Transfomers.AllowResourceVersions.
+  """
+
+  use Spark.Dsl.Transformer
+  alias Spark.Dsl.Transformer
+
+  def before?(_), do: false
+
+  def transform(dsl_state) do
+    {:ok,
+     Transformer.set_option(
```

Comment:
We need to pass the current mfa:
```elixir
Transformer.set_option(
    dsl_state,
        [:resources],
        :allow,
        {AshPaperTrail, :allow_resource_versions, [Ash.Api.Info.allow(dsl_state)]}
      )}
```

Then in that function, if there is an existing mfa it would do the logic of `version_resource || existing_mfa`.